### ISSUE TITLE
[web] Shadows are too hard

### ISSUE THREAD

**Issue by cbenhagen** [flutter/flutter#52734]
/cc @yjbanov 

Maybe a duplicate of https://github.com/flutter/flutter/issues/52477, https://github.com/flutter/flutter/issues/32215 or connected to https://github.com/flutter/flutter/issues/51237?

<img width="1392" alt="Screenshot 2020-03-17 at 14 43 53" src="https://user-images.githubusercontent.com/814785/76862041-c7b50d00-685d-11ea-8341-2e80a75a916b.png">

<img width="1392" alt="Screenshot 2020-03-17 at 14 39 27" src="https://user-images.githubusercontent.com/814785/76861613-26c65200-685d-11ea-8e91-45fb3386482c.png">

5e74b004b8cb8b8ed55030a4b7fa0a1a96575737 is the first bad commit

```
commit 5e74b004b8cb8b8ed55030a4b7fa0a1a96575737
Author: Jason Simmons <jason-simmons@users.noreply.github.com>
Date:   Wed Mar 11 12:48:50 2020 -0700

    Manual roll engine 0e2c8feeeb47..6d8611d0bd93 (31 commits) (#52431)

    2020-03-11 skia-flutter-autoroll@skia.org Roll src/third_party/dart 4093d08271f6..37530145ff53 (4 commits) (flutter/engine#17090)
    2020-03-11 skia-flutter-autoroll@skia.org Roll src/third_party/skia bf355123ae3b..0340292972b9 (9 commits) (flutter/engine#17089)
    2020-03-11 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/mac-amd64 from r_oCI... to 0Z8VF... (flutter/engine#17087)
    2020-03-11 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/linux-amd64 from v32mJ... to X3Xm2... (flutter/engine#17086)
    2020-03-11 chinmaygarde@gmail.com Remove the unused method on iOS surface to make the resource context current. (flutter/engine#17084)
    2020-03-11 chinmaygarde@gmail.com Revert "Add support for the Metal backend on all iOS builds. (#17080)" (flutter/engine#17088)
    2020-03-11 skia-flutter-autoroll@skia.org Roll src/third_party/dart ace1d9b9213a..4093d08271f6 (12 commits) (flutter/engine#17082)
    2020-03-11 chinmaygarde@google.com Add support for the Metal backend on all iOS builds. (flutter/engine#17080)
    2020-03-11 skia-flutter-autoroll@skia.org Roll src/third_party/skia d3f67dbf9f36..bf355123ae3b (9 commits) (flutter/engine#17079)
    2020-03-11 chinmaygarde@google.com Disable Embedder11yTest::A11yTreeIsConsistent to unblock LUCI. (flutter/engine#17081)
    2020-03-10 chinmaygarde@google.com Gather demangled stack traces and report the same to console on crashes. (flutter/engine#16450)
    2020-03-10 chinmaygarde@google.com Implement asynchronous texture uploads when using the Metal backend on iOS. (flutter/engine#17046)
    2020-03-10 skia-flutter-autoroll@skia.org Roll src/third_party/dart 97674262bc29..ace1d9b9213a (14 commits) (flutter/engine#17078)
    2020-03-10 egarciad@google.com Add RTree to flow (flutter/engine#16923)
    2020-03-10 skia-flutter-autoroll@skia.org Roll src/third_party/skia 78dac6dcb222..d3f67dbf9f36 (6 commits) (flutter/engine#17072)
    2020-03-10 flar@google.com Revert "Fix bounds of image_filter_layer (#16960)" (flutter/engine#17074)
    2020-03-10 chinmaygarde@google.com Use the ELF loader to setup AOT symbols in benchmark runner. (flutter/engine#17051)
    2020-03-10 skia-flutter-autoroll@skia.org Roll src/third_party/skia 23899c64e3db..78dac6dcb222 (19 commits) (flutter/engine#17069)
    2020-03-10 aam@google.com Roll dart to 97674262bc29447dc59d5c93024b18b27d4bcf98. (flutter/engine#17067)
    2020-03-10 ferhat@gmail.com [web] Fixes for Firefox & Safari double underline decoration bugs. (flutter/engine#16994)
    2020-03-10 dnfield@google.com Avoid capturing this unsafely in MultiFrameCodec (flutter/engine#16824)
    2020-03-10 yjbanov@google.com Revert "Revert "fix shadows and mask filter blurs (#16963)" (#17008)" (flutter/engine#17040)
    2020-03-10 ferhat@gmail.com Add support for firefox mac installer. Update web_ui pubspec for http.wq (flutter/engine#17044)
    2020-03-09 yjbanov@google.com fix "TREE INCONSISTENT" noise in compositing_test.dart (flutter/engine#16995)
    2020-03-09 yjbanov@google.com Add more child lifecycle tests (flutter/engine#16689)
    2020-03-09 chris@bracken.jp Add libfreetype6-dev to desktop Linux dependencies (flutter/engine#17020)
    2020-03-09 liyuqian@google.com Disable shell benchmarks (flutter/engine#17038)
    2020-03-09 flar@google.com Fix bounds of image_filter_layer (flutter/engine#16960)
    2020-03-09 liyuqian@google.com Record fml and shell benchmarks (flutter/engine#16991)
    2020-03-09 skia-flutter-autoroll@skia.org Roll src/third_party/skia c56950442dd1..23899c64e3db (11 commits) (flutter/engine#17033)
    2020-03-09 yjbanov@google.com use commit date instead of author date (flutter/engine#17032)

 bin/internal/engine.version | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
```

```
[✓] Flutter (Channel master, v1.15.23-pre.12, on Mac OS X 10.15.3 19D76, locale en-CH)
    • Flutter version 1.15.23-pre.12 at /Users/ben/flutter
    • Framework revision 3fd6808b44 (11 hours ago), 2020-03-16 22:21:01 -0400
    • Engine revision 2faadc71b6
    • Dart version 2.8.0 (build 2.8.0-dev.14.0 810d85d4cf)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/ben/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.41.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.7.1

[✓] Connected device (3 available)
    • macOS      • macOS      • darwin-x64     • Mac OS X 10.15.3 19D76
    • Chrome     • chrome     • web-javascript • Google Chrome 80.0.3987.132
    • Web Server • web-server • web-javascript • Flutter Tools

• No issues found!
```
